rfcs/text/0001-wildcard-data-type.md

```diff
@@ -115,56 +216,63 @@ Stage 2: Identifies scope of impact of changes. Are breaking changes required? S
 -->
 
-ECS-compliant data consumers should not be impacted due to the `keyword` field family[2], however ECS-compliant producers will be required for their field mappings to use `wildcard` where defined by the spec. The ECS project will need to evaluate how best to
-evolve its tooling to support `wildcard`; wildcard will be the first time a Elastic licensed feature is specified in ECS.
-
 ### Ingestion
 
-Any component producing data (Beats, Logstash, third-party developed, etc.) will need to comply with the new mappings.
+Any component producing data (Beats, Logstash, third-party developed, etc.) will need to adopt the mappings in their index templates. The discussion around the handling of OSS vs. Basic field types between OSS and Basic modules/plugins will be handled outside of this proposal.
 
 ### Usage
 
-Part of the 7.9 release is the introduction of the first field family[2], `keyword`. Grouping field types by family is intended to eliminate backwards compatibility issues when replacing an old field with a new specialized type on time-based indices (e.g. `keyword` replaced
-with `wildcard`). The `wildcard` data type will return `keyword` in the output of the field capabilities API call, and this change will enable both types to behave identically at query time. This feature eliminates concerns arising from Kibana's field compatibility checks
-in index patterns.
+Part of the 7.9 release is the introduction of the first field family[2], `keyword`. Grouping field types by family is intended to eliminate backwards compatibility issues when replacing an old field with a new specialized type on time-based indices (e.g. `keyword` replaced with `wildcard`). The `wildcard` data type will return `keyword` in the output of the field capabilities API call, and this change will enable both types to behave identically at query time. This feature eliminates concerns arising from Kibana's field compatibility checks in index patterns.
 
 ### ECS project
 
-ECS is and will remain an open source licensed project. However, there will be features available under the Elastic license that will benefit the user experience with the Elastic stack and solutions that have a place in the ECS specification. The ECS project's tooling will
-create an option in the tooling to generate OSS compatible mappings that continue supporting the OSS licensed features of Elastic.
+ECS is and will remain an open source licensed project. However, there will be features available under the Elastic license that will benefit the user experience with the Elastic stack and solutions that have a place in the ECS specification. The ECS project's tooling will create an option in the tooling to generate OSS compatible mappings that continue supporting the OSS licensed features of Elastic. The discussion and implementation of this functionality will take place outside this proposal.
 
 ## Concerns
```

Comment:
I think we should add one more concern to the section, perhaps even as the first subsection. A big driver for all of this effort was actually the combination of improving wildcard/regex performance, but also offering a better story around case insensitivity.

The latter is being addressed independently as a regexp query option, and afaik it's not out yet. I think this should get at least a cursory mention here.

(The rest of this comment isn't for the "concerns" section per se, but a continuation of my thoughts surrounding this)

We will probably even take case insensitivity into account in the whole strategy for adoption of `wildcard`:

* not having case insensitivity yet means `.text` multi-fields continue being relevant for a while
* removing `.text` multi-fields is a breaking change, so where this removal is applicable, it be done will be ECS 2.0, to be introduced with Elastic Stack 8.0...

Reply:
Yes I can see your point; it should be mentioned prominently.

Fact checking my understanding as I re-review past discussion (e.g. https://github.com/elastic/elasticsearch/issues/53603#issuecomment-629280978), would stating `wildcard` will be the preferred type over `keyword` for fields needing frequent case-insensitive queries be accurate? At least from ECS' perspective?

Reply:
I think @markharwood would be best to answer this.

The fact that this will be part of the regexp query means it'll be just as functional on `keyword` fields.

However I don't know for a fact whether it will be more performant on `wildcard`. I assume it will, since iirc the ngram backing `wildcard` is lowercased.

Reply:
Let's move forward here and assume it will. Let's capture this as a question that needs an answer during stage 2, since this is where we will firm up the precise list of fields to migrate.

Reply:
Thanks @webmat - captured.